LevelSpaceXWGUI.nlogo

```diff
@@ -302,19 +309,33 @@ to update-agent-args [a-relationship-widget ]
   ;; Ok, now we have the item. Since this is always the same, it's easy to look this up.
   let acting-entity-id agent-entity-id-from-item chosen-agent-item
 ;    let chosen-agent-entity entity-from-id run-result chosen-agent-id
-    xw:set-available-agentset-arguments get-arg-tuples acting-entity-id
+    xw:set-available-agentset-arguments get-arg-tuples-with-deps acting-entity-id acting-entity-id
   ]
 end
 
 to-report get-arg-tuples [identity-id]
+  show "arg tuples"
+  show identity-id
   let an-entity entity-from-id identity-id
   let the-args get-args an-entity
   let outer []
   foreach the-args[
-;    let tuple (list ? map [table:get last ? "name"] get-eligible-arguments an-entity)
-;    print ?
-;    print get-eligible-arguments an-entity
-    let tuple (list ? map [(word first ? ":" table:get last ? "name")] get-eligible-arguments an-entity)
+    let tuple (list ? map [(word table:get last ? "model"":" table:get last ? "name")] get-eligible-arguments an-entity)
+    set outer lput tuple outer
+    xw:set-height xw:height + 20
+  ]
+  report outer
+  
+end
+
+
+to-report get-arg-tuples-with-deps [identity-id-elig identity-id-args ]
+  let eligbility-entity entity-from-id identity-id-elig
```

Comment:
I'm glad to see we are coming together to focus on the important things that matter!